### ISSUE TITLE
iOS <=10 Localstorage Private Browsing Issue

### ISSUE THREAD

**Issue by kurtisdavis** [okta/okta-auth-js#117]
Hello,

We have discovered that ios10 and lower in private browsing has issues correctly detecting if localStorage, sessionStorage is available or not. 

```javascript
storageUtil.getHttpCache = function() {
  if (storageUtil.browserHasLocalStorage()) {
    return storageBuilder(storageUtil.getLocalStorage(), config.CACHE_STORAGE_NAME);
  } else if (storageUtil.browserHasSessionStorage()) {
    return storageBuilder(storageUtil.getSessionStorage(), config.CACHE_STORAGE_NAME);
  } else {
    return storageBuilder(storageUtil.getCookieStorage(), config.CACHE_STORAGE_NAME);
  }
};

storageUtil.getLocalStorage = function() {
  return localStorage;
};
```

iOS10 getLocalStorage, getSessionStorage returns true in private mode, but you can not write to it.  This ends up breaking the whole library.

**Comment by jmelberg-okta** [comment]
Thanks for the report @kurtisdavis!

We're going to be triaging this soon. Today we only check for the **existence** of `localStorage` and/or `sessionStorage`. Instead, we _should_ be testing that `setItem` on the possible storage option is successful.

**Comment by jmelberg-okta** [comment]
This has been resolved and included in the [`@okta/okta-auth-js@2.1.0`](https://www.npmjs.com/package/@okta/okta-auth-js) release. Thanks again for the report!
